GoogleForCreators/web-stories-wp
# Prevent workplace shifting when selecting elements

## Summary

Prevents the canvas from shifting upwards when an element that partially sits outside the canvas is selected and fixes an issue with Safari where the layout grid did not take up the entire height of the workplace.

## Relevant Technical Choices

- Uses `preventScroll: false` to prevent shifting when focusing on the selected element
- Adds a `height: 100%` to force Safari to consider all available height as grid area (needs QA)

## Testing Instructions

1. Add any element to the page
2. Resize it so that it's fairly large then drag it to the bottom of the page so that half of it is outside.
3. Unselect it
4. Select it again. The entire workspace (page, page options, thumb carousel) should no longer shift up

---



Fixes #2304
